### PR TITLE
Add support for  sagemaker-headless-execution-driver

### PR DESCRIPTION
**Pull request by oztoprakmustafa**
*Issue #, if available:*

*Description of changes:*
This PR for adding sagemaker-headless-execution-driver to sagemaker-distribution-v0 image

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: build_artifacts/v0/v0.9/v0.9.0/cpu.env.in [hunk 1]
```diff
@@ -9,6 +9,7 @@ conda-forge::pandas[version='>=2.0.3,<3.0.0']
 conda-forge::scikit-learn[version='>=1.3.1,<2.0.0']
 conda-forge::jinja2[version='>=3.1.2,<4.0.0']
 conda-forge::matplotlib[version='>=3.7.3,<4.0.0']
+conda-forge::sagemaker-headless-execution-driver[version'>=0.0.9,']
```
Comment (just4brown) [inline build_artifacts/v0/v0.9/v0.9.0/cpu.env.in:12]: Please remove trailing comma. Should be just `[version'>=0.0.9']`

File: build_artifacts/v0/v0.9/v0.9.0/gpu.env.in [hunk 2]
```diff
@@ -9,6 +9,7 @@ conda-forge::pandas[version='>=2.0.3,<3.0.0']
 conda-forge::scikit-learn[version='>=1.3.1,<2.0.0']
 conda-forge::jinja2[version='>=3.1.2,<4.0.0']
 conda-forge::matplotlib[version='>=3.7.3,<4.0.0']
+conda-forge::sagemaker-headless-execution-driver[version'>=0.0.9,']
```
Comment (just4brown) [inline build_artifacts/v0/v0.9/v0.9.0/gpu.env.in:12]: Same here regarding comma

File: test/test_artifacts/v0/sagemaker-headless-execution-driver.test.Dockerfile [hunk 3]
```diff
@@ -0,0 +1,8 @@
+ARG COSMOS_IMAGE
+FROM $COSMOS_IMAGE
+
+ARG MAMBA_DOCKERFILE_ACTIVATE=1
+
+
+# Execute the unit tests for sagemaker-headless-execution-driver
```
Comment (just4brown) [inline test/test_artifacts/v0/sagemaker-headless-execution-driver.test.Dockerfile:7]: Nit: statement below is just an import; not executing any tests. Are there unit tests in the package we can run from here?

**Review by navinsoni — APPROVED** [review]

**Review by just4brown — CHANGES_REQUESTED** [review]

File: build_artifacts/v0/v0.9/v0.9.0/cpu.env.in (hunk 1 above)
Comment (oztoprakmustafa) [reply build_artifacts/v0/v0.9/v0.9.0/cpu.env.in:12]: ack

File: test/test_artifacts/v0/sagemaker-headless-execution-driver.test.Dockerfile (hunk 3 above)
Comment (oztoprakmustafa) [reply test/test_artifacts/v0/sagemaker-headless-execution-driver.test.Dockerfile:7]: Yes there are unit test. I will fix it

File: build_artifacts/v0/v0.9/v0.9.0/gpu.env.in (hunk 2 above)
Comment (oztoprakmustafa) [reply build_artifacts/v0/v0.9/v0.9.0/gpu.env.in:12]: will fix

**Review by just4brown — APPROVED** [review]

**Review by navinsoni — APPROVED** [review]

**Review by just4brown — APPROVED** [review]

**Pull request merged**